src/Symfony/Component/Serializer/Mapping/Loader/AnnotationLoader.php

```diff
@@ -56,16 +56,21 @@ public function loadClassMetadata(ClassMetadataInterface $classMetadata): bool
         $reflectionClass = $classMetadata->getReflectionClass();
         $className = $reflectionClass->name;
         $loaded = false;
+        $classGroups = [];
 
         $attributesMetadata = $classMetadata->getAttributesMetadata();
 
         foreach ($this->loadAnnotations($reflectionClass) as $annotation) {
-            if ($annotation instanceof DiscriminatorMap) {
-                $classMetadata->setClassDiscriminatorMapping(new ClassDiscriminatorMapping(
-                    $annotation->getTypeProperty(),
-                    $annotation->getMapping()
-                ));
-            }
+            match (true) {
```

Comment:
I'm not sure using match here makes things clearer. "if" is fine to me.

Reply:
@nicolas-grekas changed it back https://github.com/symfony/symfony/pull/49594/commits/f0e6f35cf38a31d45f99dfe3b8162603544c6924